jwst/stpipe/pipeline.py

```diff
@@ -189,7 +189,8 @@ def get_config_from_reference(cls, dataset, observatory=None):
         try:
             ref_file = crds_client.get_reference_file(dataset,
                                                       pars_model.meta.reftype,
-                                                      observatory=observatory)
+                                                      observatory=observatory,
+                                                      asn_exptypes=['science'])
```

Comment:
How will this affect steps/pipelines where you WANT to load non-science exptypes, such as 'background'?

Reply:
Depends on what is meant by "load". First, a discussion of context to ensure clarity: This is only concerned with determination of the `pars-step` parameter file. This code does not affect any other reference file retrieval.

Next, if a step/pipeline takes multiple inputs, only the first input is considered. And here, multiple inputs means multiple arguments: strun step input1.fits input2.fits. The presumption is that the first input is the primary science input.

Next, if that primary science input is an association, then the first "science" member of that association is used. This is what the code change in this PR is assuring. This presumption is made because if a step/pipeline is working on an association, there must be at least one member marked "science". As far as I know, all pipelines/steps require at least one "science" member.

Now, if by "loading non-science exptypes", that means there is a step/pipeline that, when given an association that has no "science" members, will operate correctly, then yes, this will not work. However, if this is a desired feature, what is allowed can be a class parameter of the step.

If the input is just a FITS data file, such as an exposure file, then there is no association, there is no check of member type, and the input is simply used.

Reply:
OK, I didn't understand that this was just affecting the calls to CRDS to check for ref files. I thought it was the operation that was loading all of the necessary inputs to be passed around from step to step within a pipeline.
So the problem with the source catalog file getting loaded was actually an attempt by CRDS to try to treat it as a "science" file and look for matching reference files?